### ISSUE TITLE
Add quote_via keyword argument to urlencode

### ISSUE THREAD

**Issue by drnextgis** [Pylons/pyramid#3087]
How about adding `quote_via` keyword argument to `pyramid.encode.urlencode` to allow the caller to specify the encoding function as it was made in Python 3? See [#13866](https://bugs.python.org/issue13866). Something like [this](https://github.com/drnextgis/pyramid/commit/4e7a2eba304567c004b4f059740285928939d1d1):

```
>>> from pyramid.encode import url_quote, urlencode
>>> urlencode({'k': 'some value'})
'k=some+value'
>>> urlencode({'k': 'some value'}, quote_via=url_quote)
'k=some%20value'
```

**Comment by mmerickel** [comment]
Thanks @drnextgis. Would you like to provide your patch in the form of a PR with your name in CONTRIBUTORS.txt?

**Comment by mmerickel** [comment]
Closed via #3088.
